ethereum/executionclient/src/integration-test/java/tech/pegasys/teku/ethereum/executionclient/web3j/Web3JExecutionEngineClientTest.java

```diff
@@ -197,7 +203,49 @@ public void exchangeCapabilities_shouldBuildRequestAndResponseSuccessfully() thr
     assertThat(requestData.get("params")).asInstanceOf(LIST).containsExactly(consensusCapabilities);
   }
 
-  private void mockSuccessfulResponse(String responseBody) {
+  @TestTemplate
+  @SuppressWarnings("unchecked")
+  public void newPayloadV3_shouldBuildRequestAndResponseSuccessfully() {
+    assumeThat(specMilestone).isGreaterThanOrEqualTo(DENEB);
+
+    final Bytes32 latestValidHash = dataStructureUtil.randomBytes32();
+    final PayloadStatus payloadStatusResponse =
+        PayloadStatus.valid(Optional.of(latestValidHash), Optional.empty());
+
+    final String bodyResponse =
+        "{\"jsonrpc\": \"2.0\", \"id\": 0, \"result\":"
+            + "{ \"status\": \"VALID\", \"latestValidHash\": \""
+            + latestValidHash
+            + "\", \"validationError\": null}}";
+
+    mockSuccessfulResponse(bodyResponse);
+
+    final ExecutionPayload executionPayload = dataStructureUtil.randomExecutionPayload();
+    final ExecutionPayloadV3 executionPayloadV3 =
+        ExecutionPayloadV3.fromInternalExecutionPayload(executionPayload);
+    final List<VersionedHash> blobVersionedHashes = dataStructureUtil.randomVersionedHashes(3);
+
+    final SafeFuture<Response<PayloadStatusV1>> futureResponse =
+        eeClient.newPayloadV3(executionPayloadV3, blobVersionedHashes);
+
+    assertThat(futureResponse)
+        .succeedsWithin(1, TimeUnit.SECONDS)
+        .matches(
+            response ->
+                response.getPayload().asInternalExecutionPayload().equals(payloadStatusResponse));
+
+    final Map<String, Object> requestData = takeRequest();
+    verifyJsonRpcMethodCall(requestData, "engine_newPayloadV3");
+
+    assertThat(((List<Object>) requestData.get("params")).get(1))
```

Comment:
annoying but I think we could check `((List<Object>) requestData.get("params")).get(0)` as well, derived from executionPayloadV3

Reply:
Yeah, makes sense. Will tackle tomorrow.

Reply:
Actually just done it.